src/main/java/edu/aau/groupc/canteenbackend/user/controllers/UserController.java

```diff
@@ -0,0 +1,43 @@
+package edu.aau.groupc.canteenbackend.user.controllers;
+
+import edu.aau.groupc.canteenbackend.user.User;
+import edu.aau.groupc.canteenbackend.user.dto.UserDto;
+import edu.aau.groupc.canteenbackend.user.services.IUserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.Valid;
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+public class UserController {
```

Comment:
For input to be validated automatically, the controller needs to be annotated with `@Validated`.

Reply:
Nice to know. I thought `@Valid` on the `UserDto` is enough. While testing manually (POST request) it still worked for me.

Reply:
I am also not sure how SonarCloud gate works. Is it measuring only the code I try to merge, or the whole code (including already merged code)?

Reply:
I just looked it up again and **I am indeed wrong here**. `@Valid` is sufficient for our purposes. `@Validated` basically just adds support for different validation groups. Moe on that [here](https://stackoverflow.com/a/52092984) if interested. 
Thank you for pointing that out!